challenges-14-plus.yml

```diff
@@ -0,0 +1,472 @@
+    our transition, if we make one, will always take the time it takes for the Ajax call
+    to finish. This makes it hard for us to perform complex transitions when we can't
+    *pause* the render until our animation properly finishes.
+  features:
+    multiple_chopice:
+      a: |
+        Because there are no good event handlers to use when transitioning pages for
+        this particular purpose.
+      b: |
+        Actually, you *can* easily implement page transitions between clicks in 7.0.0 beta.
+      c: |
+        Because wnen a page finishes loading, Turbo replaces the `<body>` element
+        immediately, leaving no time for a transition.
+      d: |
+        Because there are other libraries that can achieve this.
+    correct: c
+
+transition_finish:
+  question: |
+    Our lead designer is *ANGRY*! When our page went live, we've discovered that some
+    of our transitions didn't work as expected! Instead of fading to red and then back
+    to white, they faded from red to white twice!
+    
+    You can imagine the anger of our clients when they didn't get to properly experience
+    all those shades of pink.
+    
+    Can you tell me exactly when did this happen and why?
+  explanation: |
+    Our `turbo:before-render` and `turbo:render` events are flawed! They happen when a real
+    page, as well as a preview page, renders!
+    
+    This is the cause of the quirkiness that we've experienced. When the user travels to
+    a page that has a preview, our `turbo-loading` class is added and removed twice,
+    causing the double fade-in from red to white!
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          This happens when the user visits a page that has been previewed before,
+          because the Turbo events `turbo:before-render` and `turbo:render` fire
+          both for real pages and preview pages alike.
+        b: |
+          This happens when the user visits a new page that has no preview, because
+          in that case, when a preview is being made, the Turbo events `turbo:before-render`
+          and `turbo:render` are fired twice.
+        c: |
+          This happens when the user visits a page by clicking the back or forward
+          button in the browser, because the Turbo event `turbo:visit` fires twice
+          in such instances.
+        d: |
+          This happens when the user clicks on a button to submit a form, because
+          the Turbo event `turbo:visit` fires twice in such instances.
+      correct: a
+
+transition_finish_2:
+  question: |
+    In order to fix this page transition issue (and save our jobs), we need to find
+    a way to identify when a `turbo:before-render` or a `turbo:render` event come
+    from a preview or a real page.
+    
+    We need to make sure our previews have the `turbo-loading` class in their `<body>`
+    elements so that the red doesn't flash twice!
+    
+    How can we do this?
```

Comment:
Maybe we can make this questions more generic (and not related to fixing some specific issue). The question is:

> From inside of JavaScript, how could we determine if the page is currently rendering a *preview* or a real page?